packages/design-system-react/src/components/avatar-icon/AvatarIcon.tsx

```diff
@@ -22,31 +22,27 @@ export const AvatarIcon = React.forwardRef<HTMLDivElement, AvatarIconProps>(
       ...props
     },
     ref,
-  ) => {
-    const baseSize = AVATAR_ICON_TO_AVATAR_BASE_SIZE_MAP[size];
-
-    return (
-      <AvatarBase
-        ref={ref}
-        shape={AvatarBaseShape.Circle}
-        size={baseSize}
-        className={twMerge(
-          AVATAR_ICON_SEVERITY_CLASSNAME_MAP[severity].background,
-          className,
-        )}
-        {...props}
-      >
-        {iconName && (
-          <Icon
-            name={iconName}
-            size={AVATAR_ICON_SIZE_TO_ICON_SIZE_CLASSNAME_MAP[size]}
-            color={AVATAR_ICON_SEVERITY_CLASSNAME_MAP[severity].iconColor}
-            {...iconProps}
-          />
-        )}
-      </AvatarBase>
-    );
-  },
+  ) => (
+    <AvatarBase
+      ref={ref}
+      shape={AvatarBaseShape.Circle}
+      size={size}
+      className={twMerge(
+        TWCLASSMAP_AVATARICON_SEVERITY_BACKGROUNDCOLOR[severity],
+        className,
+      )}
+      {...props}
+    >
+      {iconName && (
+        <Icon
+          name={iconName}
+          size={MAP_AVATARICON_SIZE_ICONSIZE[size as AvatarIconSize]}
```

Comment:
non-blocking: The type assertion `size as AvatarIconSize` is unnecessary and should be removed. Since `size` is already properly typed as `AvatarIconSize` in the component props, this assertion is redundant and bypasses TypeScript's type checking. Type assertions (`as`) should be used sparingly as they can mask potential type issues. If we're using assertions to "make TypeScript happy," it often indicates a deeper issue with our type definitions that should be addressed directly.

```suggestion
          size={MAP_AVATARICON_SIZE_ICONSIZE[size]}
```

Reply:
There was a typescript error with this before hence the type assertion. It seems like moving things to the types file helps the error to no longer appear so i'll remove this